### DISCUSSION TITLE
PGRST106 Error when accessing public schema through API

### DISCUSSION THREAD

**Discussion by owengot** [supabase/supabase#15155] [Questions]
I have also messaged support about this and they are doing their best to help me, but posting here in case someone has an idea.

I am getting the most bizarre error when querying my database via the javascript client:

```
{
    "code": "PGRST106",
    "details": null,
    "hint": null,
    "message": "The schema must be one of the following: storage"
}

```
I have tried the following with no success:
- Checked my exposed schemas in the API settings (includes public and storage)
- Tried restarting the server
- Tried creating a new schema with grants to access existing tables and specifying this in the javascript client - same error message
- Checked grants / table privileges for 'anon' through the dashboard SQL editor which confirms anonymous user has access to the "public" schema 

`
SELECT *
FROM information_schema.table_privileges
WHERE grantee = 'anon'
`

<img width="961" alt="Screenshot 2023-06-17 at 23 45 07" src="https://github.com/supabase/supabase/assets/165058/ec73f53c-3363-461f-b730-338190986717">

It seems that the Supabase API specifically restricts access to only the "storage" schema.

One day ago, the storage was upgraded by a member of the team to enable resumable uploads - I can't help but think the two issues are linked since we have had this database running for seven months now without a problem up until now. But I simply cannot see where it is coming from.

<img width="1062" alt="Screenshot 2023-06-17 at 19 11 02" src="https://github.com/supabase/supabase/assets/165058/916006d2-ec43-4fc0-a209-e2da2fc9fef3">

![Screenshot 2023-06-17 at 18 45 36](https://github.com/supabase/supabase/assets/165058/bf2d2c81-ec41-4849-b7df-a8d3a400456a)

**Comment by steve-chavez** [comment]
>  "message": "The schema must be one of the following: storage"

That indicates PostgREST is only configured with `storage` as an exposed schema.

Try doing:

```sql
ALTER ROLE authenticator SET pgrst.db_schemas TO 'public, storage, graphql, realtime';
NOTIFY pgrst;
```
